src/extension.ts

```diff
@@ -281,13 +286,61 @@ export function activate(context: vscode.ExtensionContext): void {
         await logPointsManager.toggleLogpoint(uri);
     });
 
+    initAsyncCommand(context, 'appService.editFile', async (node: IAzureNode<FileTreeItem>) => {
+        await fileEditor.showEditor(node);
+    });
+
+    initEvent(context, 'appService.fileEditor.onDidSaveTextDocument', vscode.workspace.onDidSaveTextDocument, (doc: vscode.TextDocument) => fileEditor.onDidSaveTextDocument(context.globalState, doc));
+    context.subscriptions.push(vscode.commands.registerCommand('diagnostics.LogPoints.OpenScript', openScript));
+
     initCommand(context, 'diagnostics.LogPoints.OpenScript', openScript);
 }
 
 // tslint:disable-next-line:no-empty
 export function deactivate(): void {
 }
 
+function initEvent<T>(context: vscode.ExtensionContext, eventId: string, event: vscode.Event<T>, callback: (...args: any[]) => any) {
+    context.subscriptions.push(event(wrapAsyncCallback(eventId, (...args: any[]) => Promise.resolve(callback(...args)))));
+}
+
+function wrapAsyncCallback(callbackId, callback: (...args: any[]) => Promise<any>): (...args: any[]) => Promise<any> {
+    return async (...args: any[]) => {
```

Comment:
wrapAsyncCallback has a lot of copied code from initAsyncCommand. initAsyncCommand is supposed to call wrapAsyncCallback